plugins/modules/vmware_guest_storage_policy.py

```diff
@@ -0,0 +1,436 @@
+     - '   folder: datacenter1/vm/folder1'
+     - '   folder: /folder1/datacenter1/vm'
+     - '   folder: folder1/datacenter1/vm'
+     - '   folder: /folder1/datacenter1/vm/folder2'
+     type: str
+     required: false
+   vm_home:
+     description:
+     - A storage profile policy to set on VM Home.
+     - All values and parameters are case sensitive.
+     - At least one of C(disk) or C(vm_home) are required parameters.
+     required: false
+     type: str
+   disk:
+     description:
+     - A list of disks with storage profile policies to enforce.
+     - All values and parameters are case sensitive.
+     - At least one of C(disk) and C(vm_home) are required parameters.
+     required: false
+     type: list
+     elements: dict
+     suboptions:
+       unit_number:
+         description:
+         - Disk Unit Number.
+         - Valid values range from 0 to 15.
+         type: int
+         required: true
+       policy:
+         description:
+         - Name of the storage profile policy to enforce for the disk.
+         type: str
+         required: true
+extends_documentation_fragment:
+- community.vmware.vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Enforce storepol1 policy for disk 0 and 1 using UUID
+  community.vmware.vmware_guest_storage_policy:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    validate_certs: no
+    uuid: cefd316c-fc19-45f3-a539-2cd03427a78d
+    disk:
+      - unit_number: 0
+        policy: storepol1
+      - unit_number: 1
+        policy: storepol1
+  delegate_to: localhost
+  register: policy_status
+
+- name: Enforce storepol1 policy for VM Home using name
+  community.vmware.vmware_guest_storage_policy:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    validate_certs: no
+    name: hostname1
+    vm_home: storepol1
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+msg:
+    description: Informational message on the job result.
+    type: str
+    returned: always
+    sample: "Policies successfully set."
+changed_policies:
+    description: Dictionary containing the changed policies of disk (list of dictionaries) and vm_home.
+    type: dict
+    returned: always
+    sample: {
+        "disk": [
+            {
+                "policy": "storepol1",
+                "unit_number": 0
+            }
+        ],
+        "vm_home": "storepol1"
+    }
+'''
+
+import traceback
+from ansible.module_utils.basic import missing_required_lib
+PYVMOMI_IMP_ERR = None
+try:
+    from pyVmomi import pbm, vim
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+    PYVMOMI_IMP_ERR = traceback.format_exc()
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible_collections.community.vmware.plugins.module_utils.vmware import PyVmomi, vmware_argument_spec, wait_for_task
```

Comment:
```suggestion
from ansible_collections.community.vmware.plugins.module_utils.vmware import vmware_argument_spec, wait_for_task
```

The Lint error has occurred in CI because PyVmomi doesn't use in the module.

Reply:
Updated in commit #c1911df